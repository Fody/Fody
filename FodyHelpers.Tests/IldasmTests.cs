using System;
using ApprovalTests;
using DummyAssembly;
using Fody;
using Xunit;
using Xunit.Abstractions;

public class IldasmTests :
    XunitApprovalBase
{
    IDisposable disposable;

    [Fact]
    public void StaticPathResolution()
    {
        Assert.True(Ildasm.FoundIldasm);
    }

    public IldasmTests(ITestOutputHelper outputHelper) :
        base(outputHelper)
    {
        disposable = RuntimeNamer.BuildForRuntimeAndConfig();
    }

    [Fact]
    public void VerifyMethod()
    {
        var verify = Ildasm.Decompile(GetAssemblyPath(), "DummyAssembly.Class1::Method");
        Approvals.Verify(verify);
    }

    [Fact]
    public void Verify()
    {
        var verify = Ildasm.Decompile(GetAssemblyPath());
        Approvals.Verify(verify);
    }

    static string GetAssemblyPath()
    {
        var assembly = typeof(Class1).Assembly;

        var uri = new UriBuilder(assembly.CodeBase);
        return Uri.UnescapeDataString(uri.Path);
    }

    public override void Dispose()
    {
        disposable.Dispose();
        base.Dispose();
    }
}