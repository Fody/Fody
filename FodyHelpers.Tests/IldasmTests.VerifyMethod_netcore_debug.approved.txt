.class public auto ansi beforefieldinit DummyAssembly.Class1
extends [netstandard]System.Object
{
.method public hidebysig instance void
Method(int32 param) cil managed
{
.maxstack  3
.locals init (bool V_0)
IL_0000:  nop
IL_0001:  ldarg.0
IL_0002:  ldarg.1
IL_0003:  stfld      int32 DummyAssembly.Class1::'field'
IL_0008:  ldarg.1
IL_0009:  ldc.i4.1
IL_000a:  add
IL_000b:  starg.s    param
IL_000d:  ldarg.1
IL_000e:  ldarg.0
IL_000f:  ldfld      int32 DummyAssembly.Class1::'field'
IL_0014:  ceq
IL_0016:  stloc.0
IL_0017:  ldloc.0
IL_0018:  brfalse.s  IL_002a
IL_001a:  nop
IL_001b:  ldarg.0
IL_001c:  ldarg.0
IL_001d:  ldfld      int32 DummyAssembly.Class1::'field'
IL_0022:  ldc.i4.1
IL_0023:  sub
IL_0024:  stfld      int32 DummyAssembly.Class1::'field'
IL_0029:  nop
IL_002a:  ret
}
}
